apps/ejabberd/src/ejabberd_auth.erl

```diff
@@ -246,12 +246,18 @@ set_password(User, Server, Password) ->
 do_set_password(LUser, LServer, _) when LUser =:= error; LServer =:= error ->
     {error, invalid_jid};
 do_set_password(LUser, LServer, Password) ->
-    lists:foldl(
-      fun(M, {error, _}) ->
-              M:set_password(LUser, LServer, Password);
-         (_M, Res) ->
-              Res
-      end, {error, not_allowed}, auth_modules(LServer)).
+    case is_user_exists(LUser, LServer) of
```

Comment:
I think this check should be moved to the main function which is `set_password` unless `do_set_password` is used directly from other places.